### PR TITLE
Types: Add firebase type ref

### PR DESCRIPTION
**Pull request by ShaMan123**
https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-

Did my best to infer types from code.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -221,19 +218,16 @@ declare namespace firebaseui.auth {
     selectTenant(
         projectConfig: firebaseui.auth.ProjectConfig,
         tenantIds: string[]): Promise<firebaseui.auth.SelectedTenantInfo>;
-    // tslint:disable-next-line:no-any firebase dependency not available.
-    getAuth(apiKey: string, tenantId: string|null): any;
-    // tslint:disable-next-line:no-any firebase dependency not available.
-    startSignIn(auth: any, tenantInfo?: firebaseui.auth.SelectedTenantInfo):
+    getAuth(apiKey: string, tenantId: string | null): firebase.default.auth.Auth;
+    startSignIn(auth: firebase.default.auth.Auth, tenantInfo?: firebaseui.auth.SelectedTenantInfo):
         Promise<any>;  // tslint:disable-line
```
Comment (ShaMan123) [inline types/index.d.ts:223]: What's the return type here?

**Comment by bojeil-google** [comment]
Hey @ShaMan123, can you get this to work without the `default` namespace in `firebase.default`?
The public types are defined as `firebase.auth.UserCredential` instead of `firebase.default.auth.UserCredential`.

**Comment by ShaMan123** [comment]
That's what I thought but the compiler throws an error without `.default`

**Comment by bojeil-google** [comment]
> That's what I thought but the compiler throws an error without `.default`
> It seems a `firebase` issue

**Comment by bojeil-google** [comment]
> That's what I thought but the compiler throws an error without `.default`
> It seems a `firebase` issue

The problem is if we accept this, it could be a breaking change as it may not match the existing public types.

**Comment by ShaMan123** [comment]
I disagree. Why would it be breaking? And why would it mismatch?

**Comment by bojeil-google** [comment]
> I disagree. Why would it be breaking? And why would it mismatch?

Hmm, I take that back as the underlying types should match, but we'd want to test this to confirm. Regardless, can you open an issue in the [firebase-js-sdk](https://github.com/firebase/firebase-js-sdk) repo about the`.default` types?

**Comment by ShaMan123** [comment]
Opened issue

**Comment by ShaMan123** [comment]
What you think about defining local types at the top of the file so it can be easily maintained, so you can merge and fix later on?

**Pull request closed without merging**